store/src/main/java/com/zimbra/cs/service/admin/IssueCert.java

```diff
@@ -40,53 +42,68 @@ public class IssueCert extends AdminDocumentHandler {
    * @return {@link Element} representation of {@link
    *     com.zimbra.soap.admin.message.IssueCertResponse}
    * @throws ServiceException in case if domain could not be found, domain doesn't have
-   * PublicServiceHostname and at least one VirtualHostName,
-   * server with proxy node could not be found or domain admin doesn't have rights to deal with
-   * this domain.
+   *     PublicServiceHostname and at least one VirtualHostName, server with proxy node could not be
+   *     found or domain admin doesn't have rights to deal with this domain.
    */
   @Override
   public Element handle(final Element request, final Map<String, Object> context)
       throws ServiceException {
 
-    ZimbraSoapContext zsc = getZimbraSoapContext(context);
-
-    Provisioning prov = Provisioning.getInstance();
-    String domainId = request.getAttribute(AdminConstants.A_DOMAIN);
-    Domain domain = Optional.ofNullable(prov.get(DomainBy.id, domainId))
-        .orElseThrow(() -> ServiceException.INVALID_REQUEST(
-            "Domain with id " + domainId + " could not be found.", null));
-
-    AdminAccessControl admin = checkDomainRight(zsc, domain, AdminRights.R_getDomain);
-    String adminMail = admin.mAuthedAcct.getMail();
-
-    String chain = request.getAttribute(AdminConstants.A_CHAIN_TYPE, AdminConstants.DEFAULT_CHAIN);
-
-    String domainName = Optional.ofNullable(domain.getDomainName())
-        .orElseThrow(() -> ServiceException.FAILURE(
-            "Domain with id " + domainId + " must have domain name", null));
-    String publicServiceHostname = Optional.ofNullable(domain.getPublicServiceHostname())
-        .orElseThrow(() -> ServiceException.FAILURE(
-            "Domain " + domainName + " must have PublicServiceHostname.", null));
-    String[] virtualHostNames = Optional.ofNullable(domain.getVirtualHostname())
-        .filter(hosts -> hosts.length > 0)
-        .orElseThrow(() -> ServiceException.FAILURE(
-            "Domain " + domainName + " must have at least one VirtualHostName."));
-
-    // First release will work only on ONE proxy even if there are multiple proxy in the infrastructure.
-    Server proxyServer =
-        prov.getAllServers()
-            .stream()
+    final ZimbraSoapContext zsc = getZimbraSoapContext(context);
+
+    final Provisioning prov = Provisioning.getInstance();
+    final String domainId = request.getAttribute(AdminConstants.A_DOMAIN);
+    final Domain domain =
+        Optional.ofNullable(prov.get(DomainBy.id, domainId))
+            .orElseThrow(
+                () ->
+                    ServiceException.INVALID_REQUEST(
+                        "Domain with id " + domainId + " could not be found.", null));
+
+    final AdminAccessControl admin =
+        checkDomainRight(zsc, domain, AdminRights.R_setDomainAdminDomainAttrs);
+    final String adminMail = admin.mAuthedAcct.getMail();
+
+    final String chain =
```

Comment:
Variable name `chain` is little misleading,
```suggestion
    final String chainType =
```